connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -152,12 +166,23 @@ public void stop() {
         stopRequestedLatch.countDown();
         synchronized (this) {
             if (finishedStart)
-                task.stop();
+                tryStop();
             else
                 startedShutdownBeforeStartCompleted = true;
         }
     }
 
+    private void tryStop() {
+        if (!stopped) {
+            try {
+                task.stop();
+                stopped = true;
```

Comment:
So if `task.stop()` fails, `stopped` will not be set to true and we may call it again. Is that intentional?

Reply:
Yes, that is intentional.  I imagine that it's possible that the second attempt could clean up properly whereas the first didn't.  We catch `Throwable` so even if the second attempt doesn't cleanup properly, it will just be logged.